promise.mjs

```diff
@@ -56,6 +56,47 @@ class NPromise extends Promise {
   reduce(fn, initialValue) {
     return NPromise.reduce(this, fn, initialValue);
   }
+  asCallback(callback) {
+    if (typeof callback !== "function") {
+      return this;
+    }
+
+    this.then(
+      // Resolve handler
+      (value) => {
+        process.nextTick(() => {
```

Comment:
I'm afraid this won't work in browser, can you switch to using the `queueMicrotask` API that supports both node and browser?

Reply:
yes, will push the update shortly